app.js

```diff
@@ -1,409 +1,81 @@
- * @param {string} variable.value - The name of the variable.
- * @param {string} [variable.defaultValue] - The default value of the variable (optional).
- * @returns {string} The date template string.
- */
-const generateDateTemplate = ({ uri, value, defaultValue }) => {
-  return `
-    <span resource="${uri}" typeof="https://data.vlaanderen.be/ns/mobiliteit#Variabele">
-      <span property="dct:type" content="date"></span>
-      <span class="mark-highlight-manual" property="rdf:value" datatype="xsd:date">\${${value}}</span>
-      ${
-        !defaultValue?.length
-          ? ""
-          : `<span property="https://data.vlaanderen.be/ns/mobiliteit#standaardwaarde" datatype="xsd:date">${defaultValue}</span>`
-      }
-    </span>
-  `;
-};
-
-/**
- * Applies the mappings to the template and generates the annotated template.
- *
- * @param {string} basicTemplate - The basic template string.
- * @param {TemplateVariable[]} variables - The variables to be mapped.
- * @returns {string} The annotated template string.
- */
-export const applyTemplateMappings = (basicTemplate, variables) => {
-  let annotatedTemplate = basicTemplate;
-
-  const templateGenerators = {
-    codelist: (variable) =>
-      generateCodelistTemplate({
-        ...variable,
-        source: SPARQL_ENDPOINT,
-      }),
-    location: (variable) =>
-      generateLocationTemplate({
-        ...variable,
-        source: SPARQL_ENDPOINT,
-      }),
-    date: generateDateTemplate,
-    default: generateTextTemplate,
-  };
-  for (const variable of variables) {
-    if (variable.type === "instruction") continue;
 
-    const regex = new RegExp(`\\\${${variable.value}}`, "g");
-    const generator =
-      templateGenerators[variable.type] || templateGenerators.default;
-
-    annotatedTemplate = annotatedTemplate.replace(regex, generator(variable));
+  const updatedUris = extractInsertUris(req.body);
+  const templates = await getTemplatesAndVariables(updatedUris);
+  if (!templates.results.bindings.length) {
+    return res.send("No templates found");
   }
 
-  return annotatedTemplate;
-};
-
-/**
- * Generates an annotated array based on the templates.
- * @param {Template[]} templates - The templates to be annotated.
- * @returns {Object[]} An array containing the annotated templates.
- */
-export const generateAnnotatedArray = (templates) => {
-  return templates.map(({ uri, templateValue, variables }) => {
-    return {
-      uri,
-      annotated: applyTemplateMappings(templateValue, variables),
-    };
-  });
-};
-
-export const generateUpdateQuery = (annotatedArray) => {
-  return `
-    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
-    ${annotatedArray
-      .map(
-        (template) => `
-      DELETE WHERE {
-        GRAPH <http://mu.semte.ch/graphs/mow/registry> { 
-          <${template.uri}> ext:annotated ?template .
-        }
-      };`
-      )
-      .join(" ")}
+  await processTemplateAnnotations(templates.results.bindings);
```

Comment:
no need to await I think, just early return to release connection with delta notifier and proceed